### PR TITLE
Add support for new transit parameters in Directions and Distance Matrix

### PR DESCRIPTION
**Pull request by broady**
transit_mode and transit_routing_preference.

### CODE REVIEW

**Comment by broady** [comment]
Fixes #61

**Comment by broady** [comment]
@markmcd @domesticmouse

**Comment by markmcd** [comment]
[LGTM](http://www.lgtm.in/i/ReE1)

File: googlemaps/distance_matrix.py [hunk 1]
```diff
@@ -87,6 +99,12 @@ def distance_matrix(client, origins, destinations,
     if departure_time:
         params["departure_time"] = convert.time(departure_time)
```
Comment (markmcd) [inline googlemaps/distance_matrix.py]: Will need to add 'arrival_time' too

File: googlemaps/distance_matrix.py [hunk 2]
```diff
@@ -69,7 +81,7 @@ def distance_matrix(client, origins, destinations,
     }
 
     if mode:
-        if mode not in ["driving", "walking", "bicycling"]:
+        if mode not in ["driving", "walking", "bicycling", "transit"]:
```
Comment (markmcd) [inline googlemaps/distance_matrix.py]: We validate mode, so we should probably validate transit_mode too.

Comment (broady) [inline googlemaps/distance_matrix.py]: Ah, that's because `mode` is not validated on the server. `transit_mode` is.

Comment (broady) [inline googlemaps/distance_matrix.py]: I'll add a `NOTE`.

File: googlemaps/distance_matrix.py (hunk 1 above)
Comment (broady) [inline googlemaps/distance_matrix.py]: Done

**Comment by broady** [comment]
PTAL @markmcd

**Comment by markmcd** [comment]
:+1:

**Pull request merged**